sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/With.scala

```diff
@@ -112,6 +112,22 @@ object With {
     With(replaced(commonExprRefs), commonExprDefs)
   }
 
+  /**
+   * Helper function to create a [[With]] statement when push down filter.
```

Comment:
Since this is very specific to filter pushdown, shall we put this method in the filter pushdown rule?

Reply:
OK, wait for CI to finish running.